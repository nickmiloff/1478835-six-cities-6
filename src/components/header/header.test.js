import React from 'react';
import {render} from '@testing-library/react';
import Test from '../test/test';

import Header from './header';

describe(`Header component test`, () => {
  it(`'Header component' should render correctly with authorized user`, () => {
    const {container} = render(
        <Test initial={false}>
          <Header />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`'Header component' should render correctly with unauthorized user`, () => {
    const {container} = render(
        <Test>
          <Header />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
