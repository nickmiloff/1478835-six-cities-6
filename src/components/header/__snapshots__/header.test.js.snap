// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test 'Header component' should render correctly with authorized user 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__user-name user__name"
                >
                  mail@gmail.com
                </span>
              </a>
            </li>
            <li
              class="header__nav-item user"
            >
              <button
                class="header__nav-link header__nav-link--profile"
                data-testid="logout"
                style="background-color: transparent;"
              >
                <span
                  class="header__login"
                >
                  Logout
                </span>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header component test 'Header component' should render correctly with unauthorized user 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
